pennylane/transforms/qcut.py

```diff
@@ -287,6 +295,107 @@ def fragment_graph(graph: MultiDiGraph) -> Tuple[Tuple[MultiDiGraph], MultiDiGra
     return subgraphs, communication_graph
 
 
+def _subscripted(inp: Union[int, str], subscript: int) -> str:
+    """Returns a subscripted version of ``inp`` with an integer
+    subscript."""
+    return str(inp) + str(subscript).translate(SUB)
+
+
+def _find_new_wire(target, wires: Wires) -> int:
+    """Finds a new wire label that is not in ``wires`` based
+    upon a ``target`` label. Subscripts are used to find a
+    unique new label."""
+
+    if target not in wires:
+        return target
+
+    ctr = 1
+    subscript_map = {sub: i for i, sub in enumerate(SUBS)}
+
+    # if the target is already subscripted, reset to it's parent integer
+    if isinstance(target, str) and any(char in set(target) for char in set(SUBS)):
+        ctr = subscript_map[target[-1]]  # set counter to subscript
+        target = int(target[0])  # set target to parent integer
+        wire_func = partial(_subscripted, target)
+        new_wire = target
+    elif isinstance(target, (int, str)):
+        wire_func = partial(_subscripted, target)
+        new_wire = wire_func(ctr)
+    else:
+        wire_func = lambda ctr: ctr - 1
+        new_wire = wire_func(ctr)
```

Comment:
I'm not totally clear oi what the intention was with this. If the function is not a `str` or `int` then the `new_wire` that results from the `while` loop below will have a `int` value of the final index in `wires` i.e `len(wires) - 1`. Was this the intention?